scripts/esbuild/build.ts

```diff
@@ -10,7 +11,9 @@ async function main() {
     isWatch: !!process.argv.includes('--watch'),
   });
 
-  await Promise.all([buildCli(opts), buildSysNode(opts)]);
+  // build the sys node files first as other components depend on it
+  await buildSysNode(opts);
+  await Promise.all([buildCli(opts), buildDevServer(opts)]);
```

Comment:
what exactly in the dev server build depends on the sys-node build already being done? j/w because I'd really like to avoid having the builds depend on each other at build-time if possible, currently the approach I've been taking with the esbuild bundles so far is sort of that they all mark each other as external and then can basically independently re-build

they also currently support watch mode, which if we have `await buildSysNode` before the `Promise.all` will just only build sys-node in watch mode (since that single `Promise` will never resolve).

a general point I've found in going through our rollup-based builds to port them over is that there is _a lot_ of weird stuff in our builds, so it's also totally possible if you found something like that that we may want to remove the weirdness from both the esbuild and the rollup build at the same time

Reply:
You were totally right! I was depending on a file that should have been marked as "external". I resolved the issue and put all builds back to be run in parallel. I totally agree these builds should not depend on each other, that will only result in more complexity.